metals/src/main/scala/scala/meta/internal/metals/codelenses/SuperMethodCodeLens.scala

```diff
@@ -0,0 +1,100 @@
+package scala.meta.internal.metals.codelenses
+
+import java.util.Collections.singletonList
+import org.eclipse.{lsp4j => l}
+import scala.collection.{mutable => m}
+import scala.meta.internal.implementation.ImplementationProvider
+import scala.meta.internal.implementation.SuperMethodProvider
+import scala.meta.internal.implementation.ClassHierarchyItem
+import scala.meta.internal.implementation.TextDocumentWithPath
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.TokenEditDistance
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codelenses.SuperMethodCodeLens.LensGoSuperCache
+import scala.meta.internal.metals.codelenses.SuperMethodCodeLens.emptyLensGoSuperCache
+
+final class SuperMethodCodeLens(
+    implementationProvider: ImplementationProvider,
+    buffers: Buffers,
+    userConfig: () => UserConfiguration,
+    config: MetalsServerConfig
+) extends CodeLens {
+
+  override def isEnabled: Boolean = userConfig().superMethodLensesEnabled
+
+  override def codeLenses(
+      textDocumentWithPath: TextDocumentWithPath
+  ): Seq[l.CodeLens] = {
+    val textDocument = textDocumentWithPath.textDocument
+    val path = textDocumentWithPath.filePath
+
+    val search = implementationProvider.defaultSymbolSearch(
+      textDocumentWithPath
+    )
+    val distance =
+      TokenEditDistance.fromBuffer(path, textDocument.text, buffers)
+
+    for {
+      occurrence <- textDocument.occurrences
```

Comment:
I wonder if it would be faster to go over symbols, check if they are field or method and only as the last step find occurence?